discount/src/main/java/com/kyriba/conference/discount/api/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,29 @@
+package com.kyriba.conference.discount.api;
+
+import com.kyriba.conference.discount.service.NoSuchDiscountException;
+import org.hibernate.exception.ConstraintViolationException;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+
+/**
+ * @author Igor Lizura
+ */
+@ControllerAdvice
+public class GlobalExceptionHandler extends ResponseEntityExceptionHandler {
```

Comment:
No need for global exception handler, move methods to controller and just mark with ResponseStatus annotation without usage of response.

Reply:
fixed